tests/universal/network/cancellation.cxx

```diff
@@ -0,0 +1,228 @@
+//
+// Copyright (c) 2025 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+#include <chrono>
+#include <thread>
+
+#include "zenoh.hxx"
+
+using namespace zenoh;
+using namespace std::chrono_literals;
+
+#undef NDEBUG
+#include <assert.h>
+
+void test_get() {
+    std::cout << "test_get\n";
+
+    auto session1 = Session::open(Config::create_default());
+    auto session2 = Session::open(Config::create_default());
+    KeyExpr ke("zenoh-cpp/query/cancellation/test");
+    auto queryable = session1.declare_queryable(ke, channels::FifoChannel(3));
+    std::this_thread::sleep_for(1s);
+
+    std::cout << "Check cancel removes callbacks\n";
```

Comment:
The code below seems unrelated to callback. Do I miss something or message is wrong?